File: tests/matrix/intro_mindex/matrix_intro_mindex_basic.ml
open Optitrust
open Target
open Prelude

let _ = Run.script_cpp (fun _ ->

  !! Matrix_basic.intro_mindex (var "N") [nbMulti; cCellAccess ~base:[cVar "p"] ()];
  !! Matrix_basic.intro_mindex (var "N") [cCellWrite ~base:[cVar "p"] ~index:[cTrue] (); dLHS]; (* [cCellWrite ~base:[cVar "p"] ~index:[cVar "i"] (); dLHS]; *)

)

File: tests/matrix/intro_mindex/matrix_intro_mindex_basic.cpp
#include <optitrust.h>
#include <stdlib.h>

int main () {

  const int N = 5;
  int p[5] = {0,1,2,3,4};
  for (int i = 1; i < 5; i++){
    int a = p[i-1] + 2;
    p[i] = a;
  }
  return 0;
}
